llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -411,19 +411,19 @@ TraverseCGToFindSPIRKernels(const Function *StartingFunction) {
     }
   }
 
-  return std::move(KernelNames);
+  return {std::move(KernelNames)};
 }
 
 std::vector<StringRef> getKernelNamesUsingAssert(const Module &M) {
-  auto DevicelibAssertFailFunction = M.getFunction("__devicelib_assert_fail");
+  auto *DevicelibAssertFailFunction = M.getFunction("__devicelib_assert_fail");
   if (!DevicelibAssertFailFunction)
     return {};
 
   auto TraverseResult =
-      TraverseCGToFindSPIRKernels(DevicelibAssertFailFunction);
+      traverseCGToFindSPIRKernels(DevicelibAssertFailFunction);
 
   if (TraverseResult.hasValue())
-    return std::move(*TraverseResult);
+    return std::move(*TraverseResult); // TODO remove std::move after C++17
```

Comment:
Why after C++17? Isn't it work since C++11?

Reply:
Maybe I should remove this comment at all. My intent was the following, in C++17 copy elision is guaranteed by the Standard and there is no need in `std::move` here (before C++17 compilers do such optimization too but it was not guaranteed). Usually compilers warn about using `std::move` in return statements since it prevents RVO/NRVO optimization but not in this case, there is no warning during `sycl-post-link` compilation at least on MSVC. I've tried a small example, a function that returns a vector of verbose copyable objects, but I see no copies regardless of `std::move` using. @maksimsab Could you comment using of `std::move` here? I understand why `std::move` is used in the last return statement there https://github.com/intel/llvm/pull/5552/files#diff-7d1ae2c6e3e847852b55edea2a54fc5309328f32e56800a636a4e3d9c906948bR414 (we just fill the object of `Optional`) but  not here.